### DISCUSSION TITLE
Easier way to remove metadata from processed images

### DISCUSSION THREAD

**Discussion by ronaldbarendse** [SixLabors/ImageSharp.Web#226] [Ideas] [answered]
I've noticed that by default all metadata is retained in the processed image (which seems like a very reasonable default). This can be catastrophic for the files size though, especially if you're creating thumbnail sized images, but the metadata still includes an EXIF thumbnail and/or metadata that's bigger then the actual image data.

It's easy to nuke all metadata using the following configuration:

```c#
options.OnBeforeSaveAsync = formattedImage =>
{
    formattedImage.Image.Metadata.ExifProfile = null;

    return Task.CompletedTask;
};
```

But that would also remove any EXIF orientation, etc. from the image. Does ImageSharp support metadata categories (if not, would it make sense to add them?) and/or would you be open to have this easily configured by specifying an enum of categories to keep/remove?

**Comment by JimBobSquarePants** [comment, marked as answer]
There's some very loose (lots of "other") categorization in the [EXIF specification](https://github.com/SixLabors/ImageSharp/blob/a844c78d31eaef20467b1ae8adb4beb2988cf006/src/ImageSharp/Metadata/Profiles/Exif/DC-X008-Translation-2019-E.pdf) but that's informative and not part of any stored data. 

I don't think we'd ever try to add explicit categorization to our API. It's a lot of work to maintain with very little benefit.

If individuals want to preserve only a subset of the data they would be better off iterating an enumerated collection of tags from `ExifProfile.Values` and removing each one 

```c#
// Stored as a static collection somewhere.
ExifTag[] requiredTags = { ExifTag.Orientation };

if (formattedImage.Image.Metadata.ExifProfile != null)
{
    ExifTag[] tags = formattedImage.Image.Metadata.ExifProfile.Values.Select(x => x.Tag).ToArray();
    for (int i = 0; i < tags.Length; i++)
    {
        if (Array.IndexOf(requiredTags, tags[i]) == -1)
        {
            formattedImage.Image.Metadata.ExifProfile.RemoveValue(tags[i]);
        }
    }
}
```
